features/0454-present-proof-v2/README.md

```diff
@@ -0,0 +1,345 @@
+
+> To Do: Should the previous sentence be addressed in this version? Should it be removed?
+
+### Choreography Diagram
+
+![present proof](credential-presentation.png)
+
+> To Do: Should the Ack or Problem Report be **required** to transition the state to "done"? This is not done in v1.0, and not done in the connection-less presentation use of the v1.0 protocol.
+
+## Messages
+
+The present proof protocol consists of these messages:
+
+* `propose-presentation` - Prover to Verifier (optional) - propose a presentation or send a counter-proposal in response to a `request-presentation` message
+* `request-presentation` - Verifier to Prover - request a presentation
+* `presentation` - Prover to Verifier - provide a presentation in response to a request
+
+In addition, the [`ack`](../0015-acks/README.md) and [`problem-report`](../0035-report-problem/README.md) messages are adopted into the protocol for confirmation and error handling.
+
+The messages that include `~attach` attachments may use either form of the embedded attachment&mdash;`base64` or `json`. In the examples below, one or the other is arbitrarily used.
+
+The `~attach` array is to be used to enable a single presentation to be requested/delivered in different verifiable presentation formats. The ability to have multiple attachments must not be used to request/deliver multiple different presentations in a single instance of the protocol.
+
+### Propose Presentation
+
+An optional message sent by the prover to the verifier to initiate a proof presentation process, or in response to a `request-presentation` message when the prover wants to propose using a different presentation format or request. Schema:
+
+```json
+{
+    "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/present-proof/%VER/propose-presentation",
+    "@id": "<uuid-propose-presentation>",
+    "comment": "some comment",
+    "formats" : [
+        {
+            "attach_id" : "<attach@id value>",
+            "format" : "<format-and-version>",
+        }
+    ],
+    "proposal~attach": [
+        {
+            "@id": "<attachment identifier>",
+            "mime-type": "application/json",
+            "data": {
+                "json": "<bytes for json>"
+            }
+        }
+    ]
+}
+```
+
+Description of fields:
+
+* `comment` -- a field that provides some human readable information about the proposed presentation.
+* `formats` -- contains an entry for each `filter~attach` array entry, including an optional value of the attachment `@id` (if attachments are present) and the verifiable presentation format and version of the attachment. Accepted values for the `format` items are provided in the per format "Attachment" sections immediately below.
```

Comment:
Please replace `filter~attach` with `proposal~attach`.

I see that the previous version has [presentation-preview](https://github.com/hyperledger/aries-rfcs/tree/master/features/0037-present-proof#presentation-preview) message.
Also, I noted that issue-credential-v2 has [preview-credential](https://github.com/hyperledger/aries-rfcs/tree/master/features/0453-issue-credential-v2#preview-credential) message.

Since those protocols are similar I assume that we missed something. I see two possible options:
1. We need to remove a preview-credential message from the issue-credential-v2.
2. We need to add a presentation-preview to the present-proof-v2.

Because we have attachments and formats It looks like the `1` option is what we need.
Please confirm whether I am right.

Reply:
@swcurran there does seem to be an inconsistency. Is this intentional?

Reply:
Fixed the `filter~attach` issue.  The "presentation preview" content was initially moved over to the v2 protocol under a different name.  Per @tmarkovski suggestion below, we discussed switching from this "presentation preview" format to just using an Indy "presentation-request" data structure (same as for the request-presentation message.  This was discussed on the Aries WG call on 2020.06.24 and all agreed that's what we should do.  So, "presentation preview" is now gone in both name and content, replaced in the proposal message with the presentation request.

I think that resolves this question.  Note that we definitely don't want to remove the preview credential from the issue-credential propose message.

Reply:
@swcurran "`filter~attach`" is still there, ie. the line pointed out by @soluchok has not been fixed.